### ISSUE TITLE
Duplicates on listitem rules

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#1013]
I noticed this little quirk when playing around with the `dl > div > dd` change. The following scenario (and any like it) creates two issues in axe-core:

```html
<ul>
  <div><li>Some item</li></div>
</ul>
```

The first issue is that the `li` doesn't have a `ul` parent, the second is that the `ul` has children that aren't `li` elements. I don't think axe-core should be reporting both issues. Reporting just the `ul` avoids some noise in the results. In updating the `dl > div > dd` thing, I need to make a change to how the definition-list rule deals with descending content anyway, so I'll make the change that dd / dt / li require a dl / ol / ul ancestor, rather than a direct parent. We'll still test all the same issues, but with fewer issues reported.

**Comment by dylanb** [comment]
I wonder whether this should be implemented as a "filtering layer" that sits above the core rules? We could add a bunch of different filters - e.g. look for color contrast rules that fail with the same foreground and background colors.
